analyzers/tests/SonarAnalyzer.UnitTest/MetadataReferences/NuGetMetadataFactory.Package.cs

```diff
@@ -160,34 +141,54 @@ private IEnumerable<string> SortedPackageFolders()
                     : Enumerable.Empty<string>();
             }
 
+            private async Task<string> GetLatestVersion()
+            {
+                var latest = SortedPackageFolders().Select(x => Path.GetFileName(x).Substring(Id.Length + 1)).LastOrDefault(x => char.IsNumber(x[0]));
+                if (latest == null || IsCheckForLatestPackageRequired())
+                {
+                    var resource =  await GetNuGetRepository();
+                    var versions = await resource.GetAllVersionsAsync(
+                        Id,
+                        new SourceCacheContext(),
+                        NullLogger.Instance,
+                        CancellationToken.None);
+                    return versions.OrderByDescending(x => x.Version).First(x => !x.IsPrerelease).OriginalVersion;
+                }
+
+                return latest;
+            }
+
             private bool IsCheckForLatestPackageRequired()
             {
                 // Install new nugets only once per day to improve the performance when running tests locally.
                 var nextCheck = NextCheckFilePath() is { } filePath && File.Exists(filePath) && DateTime.TryParse(File.ReadAllText(filePath), out var timestamp)
-                        ? timestamp
-                        : DateTime.MinValue;
+                    ? timestamp
+                    : DateTime.MinValue;
                 LogMessage($"Next check for latest NuGets: {nextCheck}");
                 return nextCheck < DateTime.Now;
             }
 
             private void WriteNextCheckTime()
             {
-                const int VersionCheckDays = 5;
+                const int versionCheckDays = 5;
```

Comment:
I'd prefer uppercase, because it doesn't change. It looks like a variable below.